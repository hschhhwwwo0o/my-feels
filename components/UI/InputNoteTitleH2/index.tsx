import React, { Dispatch, FunctionComponent, SetStateAction } from "react";

interface IInputNoteTitleH2 {
  value?: string;
  setValue?: Dispatch<SetStateAction<string>>;
}

const InputNoteTitleH2: FunctionComponent<IInputNoteTitleH2> = ({ value = "", setValue = () => {} }) => {
  return (
    <input
      value={value}
      onChange={e => {
        setValue(e.target.value);
      }}
      type="text"
      className="text-lg lg:text-2xl bg-transparent font-semibold outline-none mt-6 text-[#242424] dark:text-[#ECECEC] placeholder-[#bebebe] dark:placeholder-[#545454]"
      placeholder="Enter Title"
    />
  );
};

export default InputNoteTitleH2;
