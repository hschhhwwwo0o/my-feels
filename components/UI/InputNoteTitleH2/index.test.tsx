import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import InputNoteTitleH2 from "./index";

test("Test InputNoteTitleH2 component", () => {
  render(
    <span data-testid="block">
      <InputNoteTitleH2 value="" setValue={() => ""} />
    </span>
  );
  const el: HTMLElement = screen.getByTestId("block");
  expect(el).toBeInTheDocument();
  expect(el).toMatchSnapshot();
});
