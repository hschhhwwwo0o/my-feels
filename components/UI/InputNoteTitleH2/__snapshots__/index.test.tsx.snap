// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test InputNoteTitleH2 component 1`] = `
<span
  data-testid="block"
>
  <input
    class="text-lg lg:text-2xl bg-transparent font-semibold outline-none mt-6 text-[#242424] dark:text-[#ECECEC] placeholder-[#bebebe] dark:placeholder-[#545454]"
    placeholder="Enter Title"
    type="text"
    value=""
  />
</span>
`;
